src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/SmppMessagingHandler.java

```diff
@@ -21,78 +33,120 @@
+                addCharacteristic(message, ParameterNames.ERROR_CODE, ErrorCodes.ERROR_SYSTEM_GENERAL_INTERNAL_SERVER_ERROR);
             }
         }
         return message;
     }
 
+    private void addCharacteristic(CommunicationMessage message, String name, String value) {
+        message.getCharacteristic().add(new CommunicationRequestCharacteristic()
+            .name(name)
+            .value(value));
+    }
+
     @Override
     public CommunicationMessage handle(CommunicationMessageCreate messageCreate) {
         CommunicationMessage communicationMessage = mapper.messageCreateToMessage(messageCreate);
         this.handle(communicationMessage);
         return communicationMessage;
     }
 
-    private void sendSmppMessage(CommunicationMessage message, Messaging messaging,
-                                 String phoneNumber) throws Exception {
+    private String sendSmppMessage(CommunicationMessage message, Messaging messaging,
+                                   String phoneNumber) throws Exception {
         RuleResponse ruleResponse = businessRuleValidator.validate(message);
         if (!ruleResponse.isSuccess()) {
-            failMessage(message, ruleResponse.getResponseCode(), ERROR_BUSINESS_RULE_VALIDATION);
-            return;
+            throw new RuleValidationException(ruleResponse);
         }
+
         String messageId = smppService.send(phoneNumber, message.getContent(), message.getSender().getId(),
-            getDeliveryReport(message.getCharacteristic()));
+            getDeliveryReport(message.getCharacteristic()), buildOptionalParameters(message),
+            getFromCharacteristics(message.getCharacteristic(), ParameterNames.VALIDITY_PERIOD, Ints::tryParse),
+            getFromCharacteristics(message.getCharacteristic(), ParameterNames.PROTOCOL_ID, Ints::tryParse)
+        );
         String queueName = messaging.getSentQueueName();
         sendMessage(success(messageId, message), queueName);
-        log.info("Message success sent to {}", queueName);
+        log.info("Message success sent to {}, messageId {}", queueName, messageId);
+        return messageId;
+    }
+
+    private Map<Short, String> buildOptionalParameters(CommunicationMessage message) {
+        return Optional.ofNullable(message.getCharacteristic())
+            .map(this::collectOptionalParameters)
+            .orElse(Collections.emptyMap());
+    }
+
+    private Map<Short, String> collectOptionalParameters(List<CommunicationRequestCharacteristic> characteristics) {
+        return characteristics.stream()
+            .filter(Objects::nonNull)
+            .filter(c -> Objects.nonNull(c.getName()))
+            .filter(ch -> ch.getName().startsWith(ParameterNames.OPTIONAL_PARAMETER_PREFIX))
+            .collect(Collectors.toMap(
+                ch -> Short.parseShort(ch.getName().substring(ParameterNames.OPTIONAL_PARAMETER_PREFIX.length())),
+                CommunicationRequestCharacteristic::getValue));
```

Comment:
@VictorKhoroshko 
I see you have avoided inner lambdas with this refactoring. 
And i'd like to suggest you to refactor thiso code a little bit more into functional style:
```
    private Map<Short, String> buildOptionalParameters(CommunicationMessage message) {
        return Optional.ofNullable(message.getCharacteristic())
                       .stream()
                       .flatMap(Collection::stream)
                       .filter(isOptionalParameter())
                       .collect(Collectors.toMap(SmppMessagingHandler::parseParamAsShort,
                                                 CommunicationRequestCharacteristic::getValue));
    }

    private Predicate<CommunicationRequestCharacteristic> isOptionalParameter(){
        return c -> c != null && StringUtils.startsWith(c.getName(), OPTIONAL_PARAMETER_PREFIX);
    }

    private static Short parseParamAsShort(CommunicationRequestCharacteristic ch) {
        return Short.parseShort(ch.getName().substring(OPTIONAL_PARAMETER_PREFIX.length()));
    }
    ``` 
    IMHO in this case the stream processing logic in `buildOptionalParameters` method is easier to read.